### PR TITLE
chore: Replace availability and nextAvailability with useAvailability and useNextAvailability composables [CW-3472]

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description

Please include a summary of the change and issue(s) fixed. Also, mention relevant motivation, context, and any dependencies that this change requires.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/widget/composables/useNextAvailability.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
+import { ref, computed } from 'vue';
```
Comment (muhsin-k) [inline app/javascript/widget/composables/useNextAvailability.js:1]: @fayazara Can you write the specs for the the composition `useNextAvailability.js` or is all the cases already covered in `useAvailability.js`?

File: app/javascript/widget/mixins/specs/useAvailabilityComposable.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { describe, it, expect, beforeEach, vi } from 'vitest';
```
Comment (muhsin-k) [inline app/javascript/widget/mixins/specs/useAvailabilityComposable.spec.js:1]: Please keep this spec under `composables` folder and remove all the mixins related to availability?

File: app/javascript/widget/composables/useAvailability.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        return t('REPLY_TIME.IN_A_DAY');
+      default:
+        return t('REPLY_TIME.IN_A_FEW_HOURS');
+    }
+  });
+
+  const replyWaitMessage = computed(() => {
+    const { workingHoursEnabled } = channelConfig.value;
+    if (workingHoursEnabled) {
+      return isOnline.value
```
Comment (iamsivin) [inline app/javascript/widget/composables/useAvailability.js:33]: There is some Eslint issue with `isOnline` computed property `'isOnline' was used before it was defined.`. Please take a look.

File: app/javascript/widget/composables/useAvailability.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      : t('TEAM_AVAILABILITY.OFFLINE');
+  });
+
+  const outOfOfficeMessage = computed(
+    () => channelConfig.value.outOfOfficeMessage
+  );
+
+  const currentDayAvailability = computed(() => {
+    const { utcOffset } = channelConfig.value;
+    const dayOfTheWeek = getDateWithOffset(utcOffset).getDay();
```
Comment (iamsivin) [inline app/javascript/widget/composables/useAvailability.js:48]: `getDateWithOffset` was used before it was defined.

File: app/javascript/widget/composables/useAvailability.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    return isOnline.value
+      ? replyTimeStatus.value
+      : t('TEAM_AVAILABILITY.OFFLINE');
+  });
+
+  const outOfOfficeMessage = computed(
+    () => channelConfig.value.outOfOfficeMessage
+  );
+
+  const currentDayAvailability = computed(() => {
```
Comment (iamsivin) [inline app/javascript/widget/composables/useAvailability.js:46]: @fayazara See if we can move this to a helper file.

File: app/javascript/widget/composables/useAvailability.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+      closedAllDay: workingHourConfig.closed_all_day,
+      openHour: workingHourConfig.open_hour,
+      openMinute: workingHourConfig.open_minutes,
+      closeHour: workingHourConfig.close_hour,
+      closeMinute: workingHourConfig.close_minutes,
+      openAllDay: workingHourConfig.open_all_day,
+    };
+  });
+
+  const isInBetweenTheWorkingHours = computed(() => {
```
Comment (iamsivin) [inline app/javascript/widget/composables/useAvailability.js:62]: See if we can move this to a helper file.

File: app/javascript/widget/composables/useAvailability.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    );
+    return isAfterStartTime && isBeforeEndTime;
+  });
+
+  const isInBusinessHours = computed(() => {
+    const { workingHoursEnabled } = channelConfig.value;
+    return workingHoursEnabled ? isInBetweenTheWorkingHours.value : true;
+  });
+
+  const getDateWithOffset = utcOffset => {
```
Comment (iamsivin) [inline app/javascript/widget/composables/useAvailability.js:104]: Let us move it outside the composable or move it to `DateHelper.js`

File: app/javascript/widget/mixins/specs/useAvailabilityComposable.spec.js
Comment (iamsivin) [inline app/javascript/widget/mixins/specs/useAvailabilityComposable.spec.js:1]: Update the file name to `useAvailability.spec.js`

File: app/javascript/widget/composables/useNextAvailability.js [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+    }
+    return hoursAndMinutesToBack.value;
+  });
+
+  /**
+   * Gets the next day of the week
+   * @param {number} day - Current day of the week
+   * @returns {number} Next day of the week
+   */
+  function getNextDay(day) {
```
Comment (iamsivin) [inline app/javascript/widget/composables/useNextAvailability.js:182]: Let us move it to a helper file or outside the composable.

File: app/javascript/widget/composables/useNextAvailability.js [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+    return (day + 1) % 7;
+  }
+
+  /**
+   * Gets the next working hour for a given day
+   * @param {number} day - Day of the week
+   * @returns {Object|null} Working hour object for the next working day, or null if not found
+   */
+  function getNextWorkingHour(day) {
```
Comment (iamsivin) [inline app/javascript/widget/composables/useNextAvailability.js:191]: See if we can move it to a helper file or outside the composable.

File: app/javascript/widget/composables/useNextAvailability.js [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+  }
+
+  /**
+   * Calculates hours and minutes until next day open
+   * @param {number} openHour - Opening hour
+   * @param {number} openMinutes - Opening minutes
+   * @param {number} closeHour - Closing hour
+   * @returns {Object} Object containing hours and minutes left
+   */
+  function getHoursAndMinutesUntilNextDayOpen(
```
Comment (iamsivin) [inline app/javascript/widget/composables/useNextAvailability.js:208]: Let us move it to a helper file.

**Review by iamsivin — COMMENTED** [review]
@fayazara Added some comments could you please take a look?

**Pull request closed without merging**